rust/src/mysql/mysql.rs

```diff
@@ -0,0 +1,1133 @@
+use std::collections::VecDeque;
+use std::ffi::CString;
+
+use nom7::IResult;
+
+use crate::applayer::*;
+use crate::conf::{conf_get, get_memval};
+use crate::core::*;
+
+use super::parser::*;
+
+pub const MYSQL_CONFIG_DEFAULT_STREAM_DEPTH: u32 = 0;
+
+static mut MYSQL_MAX_TX: usize = 1024;
+
+static mut ALPROTO_MYSQL: AppProto = ALPROTO_UNKNOWN;
+
+#[derive(FromPrimitive, Debug, AppLayerEvent)]
+pub enum MysqlEvent {
+    TooManyTransactions,
+}
+
+#[derive(Debug)]
+pub struct MysqlTransaction {
+    pub tx_id: u64,
+
+    pub version: Option<String>,
+    pub command_code: Option<u8>,
+    pub command: Option<String>,
+    pub affected_rows: Option<u64>,
+    pub rows: Option<Vec<String>>,
+    pub tls: Option<bool>,
+
+    pub complete: bool,
+    pub tx_data: AppLayerTxData,
+}
+
+impl Transaction for MysqlTransaction {
+    fn id(&self) -> u64 {
+        self.tx_id
+    }
+}
+
+impl Default for MysqlTransaction {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl MysqlTransaction {
+    pub fn new() -> Self {
+        Self {
+            tx_id: 0,
+            version: None,
+            command_code: None,
+            command: None,
+            affected_rows: None,
+            tls: None,
+            tx_data: AppLayerTxData::new(),
+            complete: false,
+            rows: None,
+        }
+    }
+}
+
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub enum MysqlStateProgress {
+    // Default State
+    Init,
+
+    // Connection Phase
+    // Server send HandshakeRequest to Client
+    Handshake,
```

Comment:
Is there a MySQL spec about these states ?

Reply:
Yes, Please check this.
https://dev.mysql.com/doc/dev/mysql-server/latest/page_protocol_connection_phase.html

Reply:
Cool, please add this link as a comment before the enum ;-)